src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1486,7 +1522,7 @@ private bool WriteSend(EndpointManager.Send send)
                         string.Format("Discarding oversized payload sent to {0}: max allowed size {1} bytes, actual size of encoded {2} was {3} bytes.",
                             send.Recipient,
                             Transport.MaximumPayloadBytes,
-                            send.Message.GetType(),
+                            LogPossiblyWrappedMessageType(send.Message),
```

Comment:
Captured all of the relevant `EndpointWriter` callsites here - unfortunately, on the `EndpointReader` side we don't know what the underlying type is because it fails during deserialization, which is where we would extract that kind of information.